### PR TITLE
jsonschemas: schema clean up

### PR DESCRIPTION
**Pull request by jmartinm**
* Removes fields only used on ElasticSearch and not part of the record
  data model.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
I have some questions for other fields. I am thinking here from the perspective of the new [record editor](https://github.com/inspirehep/record-editor) were we would use the schema as a template to fill the record so having too many separate fields seems a bit confusing.

* [ ] `spyres_sysnos` - [see schema](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L106-L113) does not seem to be [much used](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/dojson/current_marcxml_usage/hep.txt#L8782-L8818). Shall we instead use [persistent identifiers](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L972) or [external_system_numbers](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L949) ?

* [ ] [titles_old](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L149). Is this something we want to have as separate field? Do we want, for example, the editor to show that field as something to fill in? Or it would be better as part of titles (with a flag or comment saying that it is old)

* [ ] [title_variations](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L5250). Same as above, is there a good use on keeping it separate? Wouldn't it be more natural to edit everything in the `titles` field ?

* [ ] [hidden_note CDS](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/hep.json#L249) - [see usage](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/dojson/current_marcxml_usage/hep.txt#L4917) , shall we move the content the `source` subfield writing `CERN` or `CDS`

**Comment by jmartinm** [comment]
@inspirehep/inspire-content

**Comment by jacquerie** [comment]
Do `titles_old` and `title_variations` fall under #1257, which wasn't warmly welcomed by @ksachs and @fschwenn?

File: inspirehep/modules/records/jsonschemas/records/hep.json [hunk 1]
```diff
@@ -654,12 +641,6 @@
                         "description": "Just in case...",
                         "title": "Other IDs"
                     },
-                    "uuid": {
-                        "title": "UUID",
-                        "description": "Universally unique identifier of the author.",
-                        "type": "string",
-                        "pattern": "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
-                    },
```
Comment (kaplun) [inline inspirehep/modules/records/jsonschemas/records/hep.json]: Actually this should be in the schema. It should not be touched by catalogers though. This is added on `before_record_update`/`_inser`, so it goes directly into the DB.

**Comment by kaplun** [comment]
Re: #1257, as long as we preserve all titles, they are fine (e.g. we came IRL to the proposal of storing all titles as a list, where anything that is not the first title is either an alternative or older title.) The use case of old_titles is for retrievability so having them in the list without them being displayed is ok. Plus @mihaibivol says he can devise a merging algorithm that puts olders value at the end of a list.

**Comment by kaplun** [comment]
>  spyres_sysnos - see schema does not seem to be much used. Shall we instead use persistent identifiers or external_system_numbers ?

I think we can kill this. Does it has any use at all nowadays?

> hidden_note CDS - see usage , shall we move the content the source subfield writing CERN or CDS

:+1:

File: inspirehep/modules/records/jsonschemas/records/hep.json (hunk 1 above)
Comment (jacquerie) [inline inspirehep/modules/records/jsonschemas/records/hep.json]: Ok, then I'm readding a WIP tag and we can wait for @jmartinm to be back. If something else is blocked by this (I don't see how) I'll resurrect and integrate.

**Comment by jmartinm** [comment]
I will take care of this PR today for the fields without doubts and create a different issue with the pending ones.

File: inspirehep/modules/records/jsonschemas/records/hep.json (hunk 1 above)
Comment (jmartinm) [inline inspirehep/modules/records/jsonschemas/records/hep.json]: 👍

**Comment by jmartinm** [comment]
Ready for review. Have not touched titles for the moment.

File: inspirehep/dojson/common/base.py [hunk 2]
```diff
@@ -143,38 +144,41 @@ def creation_modification_date2marc(self, key, value):
...
+                    "institute": "SPIRES",
+                    "value": val.get('a'),
+                    "obsolete": True
+                }
+            )
         elif 'd' in val:
             new_recid = val.get('d')
     if new_recid is not None:
         # FIXME we are currently using the default /record API. Which might
         # resolve to a 404 response.
```
Comment (jacquerie) [inline inspirehep/dojson/common/base.py]: I believe this is no longer true?

File: inspirehep/dojson/common/base.py [hunk 3]
```diff
@@ -143,38 +144,41 @@ def creation_modification_date2marc(self, key, value):
...
         self['new_record'] = get_record_ref(new_recid)
-    return sysnos or None
 
+    self['external_system_numbers'] = external_system_numbers
 
-@hep2marc.over('970', '(spires_sysnos|new_record)')
-@hepnames2marc.over('970', '(spires_sysnos|new_record)')
+    raise IgnoreKey
```
Comment (jacquerie) [inline inspirehep/dojson/common/base.py]: Doesn't `@utils.ignore_value` at the top already achieve this?

**Comment by jacquerie** [comment]
LGTM (but I haven't tested). Left a couple of questions.

File: inspirehep/dojson/common/base.py (hunk 3 above)
Comment (jmartinm) [inline inspirehep/dojson/common/base.py]: Looking at the code of that utility it seems to be the same indeed.

**Comment by jmartinm** [comment]
Comments addressed and rebased.

**Pull request merged**